apps/dashboard/app/models/script.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class Script
+  include ActiveModel::Model
+
+  attr_reader :title
+
+  attr_reader :id
+
+  class << self
+
+    def find(id, project_dir)
+      all(project_dir).select do |script|
+        # Rails.logger.debug("#{id} == #{script.id}")
+        id == script.id
+      end.first
+    end
+
+    def all(project_dir)
+      Dir.glob("#{Project.dataroot}/#{project_dir}/.ondemand/scripts/*.yml").map do |file|
+        Script.from_yaml(file, project_dir)
+      end
+    end
+
+    def from_yaml(file, project_dir)
+      contents = File.read(file)
+      opts = YAML.safe_load(contents).to_h
+      opts.merge!({ id: File.basename(file, '.yml') })
+      opts.merge!({ project_dir: project_dir.to_s })
+      new(opts)
+    end
+    alias_method :from_yml, :from_yaml
+
+    def next_id(project_dir)
+      all(project_dir).map(&:id).map(&:to_i).max + 1
```

Comment:
Break this into more than one line so it is more obvious. 

Also, I thought we agreed the `id` was hard to write at this moment? We already closed out a PR because of that that did largely the same thing, though using a lookup file (I'm now lost on what we've decided) rather than iterating through the whole directory and doing a calculation.

Reply:
> Break this into more than one line so it is more obvious.

You're saying this is _easier_ to read?

```ruby
all(project_dir).map do |project|
  project.id
end.map do |id|
   id.to_i
end.max
```
> Also, I thought we agreed the id was hard to write at this moment? We already closed out a PR because of that that did largely the same thing, though using a lookup file (I'm now lost on what we've decided) rather than iterating through the whole directory and doing a calculation.

IDK about hard - I just did it directly.  I would have been happy with the other that we refactored afterwards, but we didn't for whatever reason.

Reply:
No. We look to be calling a method, to then call a method on the returned result, to then call a mutation on that to then find the max value and then to add an integer to that max value.

We can keep it, but I find this kind of code hard to read and maintain.

Reply:
> No. We look to be calling a method, to then call a method on the returned result, to then call a mutation on that to then find the max value and then to add an integer to that max value.

> We can keep it, but I find this kind of code hard to read and maintain.

Yea chains are very common pattern in ruby. I actually think naming temporary variables seems to be an anti-pattern in ruby.

Reply:
Oh!! I'm sorry man, I'm not being clear here. I agree, I just meant:
```
all(project_dir)
  .map(&:id)
  .map(&:to_i)
  .max + 1
```

I just find it simpler to follow is all. Sorry, I should have just recommened this to start.

Reply:
Got it! Yea, that's all good. Updated.